client/blocks/image-selector/preview.jsx

```diff
@@ -0,0 +1,236 @@
+		this.updateImageState( () => {
+			const { itemIds, siteId } = this.props;
+			if ( isEqual( this.state.images.map( image => image.ID ), itemIds ) ) {
+				return;
+			}
+
+			defer( () => {
+				itemIds.map( id => {
+					id = parseInt( id, 10 );
+					const media = MediaStore.get( siteId, id );
+					if ( ! media ) {
+						MediaActions.fetch( siteId, id );
+					}
+				} );
+			} );
+		} );
+	};
+
+	updateImageState = callback => {
+		const { itemIds, onImageChange, siteId } = this.props;
+		this.setTransientImages();
+		const images = uniq(
+			itemIds
+				.map( id => {
+					return MediaStore.get( siteId, id );
+				} )
+				.filter( function( e ) {
+					return e;
+				} )
+		);
+
+		this.setState( { images }, () => {
+			if ( 'function' === typeof callback ) {
+				callback();
+			}
+		} );
+		defer( () => {
+			const imageIds = images.map( image => image.ID );
+			if (
+				onImageChange &&
+				images &&
+				images.length === itemIds.length &&
+				! isEqual( imageIds, itemIds )
+			) {
+				onImageChange( images );
+			}
+		} );
+	};
+
+	src = image => {
+		return url( image, {
+			maxWidth: this.props.maxWidth,
+			size: 'post-thumbnail',
+		} );
+	};
+
+	renderPlaceholder = ID => {
+		let placeholder = <Spinner />;
+		if ( this.state.transientImages[ ID ] ) {
+			placeholder = <img src={ this.state.transientImages[ ID ] } alt="placeholder" />;
+		}
+		return placeholder;
+	};
+
+	renderUploaded = ( { URL, ID } ) => {
+		return (
+			<figure>
+				<ImagePreloader src={ URL } placeholder={ this.renderPlaceholder( ID ) } />
+			</figure>
+		);
+	};
+
+	renderImage = image => {
+		const src = this.src( image );
+		const { onImageClick, onRemoveImage, showEditIcon, translate } = this.props;
+		const id = image.ID || image.transientId;
+
+		const removeImage = () => {
+			onRemoveImage( image );
+		};
+
+		const classes = classNames( 'image-selector__item', {
+			'is-transient': image.transient || ! src,
+			preview: null === src,
+		} );
+
+		return (
+			<div className={ classes } key={ id }>
+				<Button
+					className="image-selector__image"
+					onClick={ onImageClick }
+					borderless
+					compact
+					data-tip-target="image-selector-image"
+				>
+					<Spinner />
+					{ src ? this.renderUploaded( image ) : <span /> }
+					{ showEditIcon && <Gridicon icon="pencil" className="image-selector__edit-icon" /> }
+				</Button>
+				<Button
+					onClick={ removeImage }
+					compact
+					aria-label={ translate( 'Remove image' ) }
+					className="image-selector__remove"
+				>
+					<Gridicon icon="cross-small" size={ 24 } className="image-selector__remove-icon" />
+				</Button>
+			</div>
+		);
+	};
+
+	renderUploadPlaceholder() {
+		const { onImageClick, translate, compact, multiple } = this.props;
+
+		const classes = classNames( 'image-selector__uploader-wrapper', {
+			compact,
+		} );
+
+		const addString = multiple ? translate( 'Add images' ) : translate( 'Add image' );
```

Comment:
Hi! I've found a possible matching string that has already been translated **57** times:
`translate( 'Add Images', { context: 'label for button in the gallery widget; should not be longer than ~13 characters long'} )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Reply:
@joshuatf could you look at those @a8ci18n warnings and act on them somehow?

Reply:
@nb Sure, but I'm a bit unsure on how to act on this.  Is there anything else needed aside from 👍or 👎?

Reply:
Ah- gotcha!  Completely missed the capital 'i' at first glance 😅

Reply:
Spoke too soon; still getting that warning.  Do we need to include the same context to suppress the warning?

Reply:
Well, the end goal is to reuse the translation, not just to suppress the warning :) And you're right, we need the context in order to reuse it. This is unfortunate, ideally this wouldn't have been context, but just a translator comment since contexts are used for disambiguation. The string [comes from WordPress core](https://core.trac.wordpress.org/browser/trunk/src/wp-includes/widgets/class-wp-widget-media-gallery.php?rev=42343#L35).